### PR TITLE
Update attrs plugin to work with new semantic analyzer

### PR DESCRIPTION
**Pull request by ilevkivskyi**

### CODE REVIEW

File: mypy/plugins/attrs.py [hunk 1]
```diff
@@ -43,6 +44,9 @@
     'attr.attr',
 }  # type: Final
 
+# TODO: should we use underscored names like for named tuples?
+SELF_TVAR_NAME = 'AT'  # type: Final
```
Comment (JukkaL) [inline mypy/plugins/attrs.py]: I'd suggest using underscore (or double underscore?) here.

File: mypy/checker.py [hunk 2]
```diff
@@ -1614,6 +1614,9 @@ def determine_type_of_class_member(self, sym: SymbolTableNode) -> Optional[Type]
         if isinstance(sym.node, TypeInfo):
             # nested class
             return type_object_type(sym.node, self.named_type)
+        if isinstance(sym.node, TypeVarExpr):
+            # No need to check type vars since they can't be used at runtime.
```
Comment (JukkaL) [inline mypy/checker.py]: This comment is a bit unclear. Do you mean that their use is rejected in an expression/runtime context by mypy?

File: test-data/unit/check-attr.test [hunk 3]
```diff
@@ -577,7 +577,10 @@ class C:
 
 [builtins fixtures/list.pyi]
 
+-- This is tricky with new analyzer:
+-- We need to know the analyzed type of a function while still processing top-level.
```
Comment (JukkaL) [inline test-data/unit/check-attr.test]: Probably worth creating an issue to track this.

File: mypy/checker.py (hunk 2 above)
Comment (ilevkivskyi) [reply mypy/checker.py]: Yes.

File: test-data/unit/check-attr.test (hunk 3 above)
Comment (ilevkivskyi) [reply test-data/unit/check-attr.test]: Opened #6438

**Pull request merged**